app/controllers/products_controller.rb

```diff
@@ -0,0 +1,66 @@
+class ProductsController < ApplicationController
+
+before_action only: [:show, :edit, :update, :retire] { @product = Product.find(params[:id]) }
+
+  def index
+    if params[:search]
+      @products = Product.search(params[:search]).where(retire: false).order(stock: :desc)
+    else
+    @products = Product.where(retire: false).order(stock: :desc)
+    end
+  end
+
+  def show
+    @review = Review.new(product_id: @product.id)
+    if @order_item = @current_order.orderitems.where("product_id = ?", params[:id]).first
```

Comment:
This is an odd if/else statement since it doesn't look like there is any block associated with the `if`